### PR TITLE
Improve cipher_pref_check()

### PR DESCRIPTION
**Pull request by dcooper16**
Some servers are configured to prioritize ChaCha ciphers if those ciphers are preferred by the client, even if the server is generally configured to use the server's cipher preferences rather than the client's. As a result of this, if a ChaCha cipher appears in the ClientHello before a non-ChaCha cipher, the server may select the ChaCha cipher even if the server is configured to prefer the non-ChaCha cipher.

In a few cases, e.g., cloudflare.com for TLS 1.2, this affects the ordering of the ciphers presented by `cipher_pref_check()`. This PR fixes the problem by having `cipher_pref_check()` (and `check_tls12_pref()`) always place any ChaCha ciphers at the end of the cipher list in the ClientHello. This ensures that `cipher_pref_check()` presents the ciphers in the server's preference order.

NOTE: After this PR has been merged, a follow-up PR will be submitted that checks whether the server is configured to prioritize ChaCha ciphers (as was requested in #1733.

### CODE REVIEW

File: testssl.sh [hunk 1]
```diff
@@ -7074,8 +7096,32 @@ cipher_pref_check() {
...
+                    ciphers_found2[nr_ciphers]=false
+                    hexcode[nr_ciphers]="${hexc:2:2},${hexc:7:2}"
+                    rfc_ciph[nr_ciphers]="${TLS_CIPHER_RFC_NAME[i]}"
+                    if [[ $proto == tls1_3 ]]; then
+                         [[ "${hexc:2:2}" == "13" ]] && nr_ciphers+=1
+                    elif [[ $proto == tls1_2 ]]; then
+                         [[ "${hexc:2:2}" != "13" ]] && nr_ciphers+=1
+                    elif [[ ! "${TLS_CIPHER_RFC_NAME[i]}" =~ SHA256 ]] && \
+                         [[ ! "${TLS_CIPHER_RFC_NAME[i]}" =~ SHA384 ]] && \
+                         [[ "${TLS_CIPHER_RFC_NAME[i]}" != *"_CCM" ]] && \
```
Comment (drwetter) [inline testssl.sh:7117]: Could you double check pls whether we need the quotes here in the RHS? (also pls following line)

**Comment by drwetter** [comment]
Hi David,

thanks. Please see the single comment. I always put trust in you that you have tested this, so I did not. So I would after resolving the minor thing merge this.

Doesn't this make sense for 3.0 too?

Cheers, Dirk

**Comment by dcooper16** [comment]
Okay, I removed the unnecessary quotes. I just made a second copy of the code that was already there and didn't notice that it contained quotes that weren't needed.

I'll work on a PR for the 3.0 branch.

**Comment by drwetter** [comment]
Thanks a lot, David

**Pull request merged**